flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -148,6 +150,17 @@ private Namespace toNamespace(String database) {
   }
 
   TableIdentifier toIdentifier(ObjectPath path) {
+    String objectName = path.getObjectName();
+    List<String> tableName = Splitter.on('$').splitToList(objectName);
+    if (tableName.size() > 1 && MetadataTableType.from(tableName.get(1)) != null) {
+      return TableIdentifier.parse(
+          toNamespace(path.getDatabaseName()).toString()
```

Comment:
I would make `toNamespace` a static method taking two params (instead of one currently)
```
/** Append a new level to the base namespace */
private static Namespace appendLevel(Namespace baseNamespace, String newLevel)
```

Then we can append a new level `tableName.get(0)` to the baseNamespace. treat `tableName.get(1)` as name when constructing the TableIdentifier.
```
public static TableIdentifier of(Namespace namespace, String name)
```

Reply:
Done. Please see if it meets your expectations.